src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.MultiAgent.cs

```diff
@@ -848,9 +798,21 @@ private static int CurlSeekCallback(IntPtr context, long offset, int origin)
                 {
                     try
                     {
-                        if (easy._requestContentStream.CanSeek)
+                        // If libcul is requesting we seek back to the beginning and if the request
+                        // content stream is in a position to reset itself, reset and let libcurl
+                        // know we did the seek; otherwise, let it know we can't seek.
+                        if (offset == 0 && origin == (int)SeekOrigin.Begin && 
```

Comment:
Is this the only way libcurl rewinds? Or can it ask for negative offset from SeekOrigin.End?